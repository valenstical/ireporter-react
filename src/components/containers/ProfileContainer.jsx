import React, { Component } from 'react';
import { connect } from 'react-redux';
import formDataJSON from 'formdata-json';
import PropTypes from 'prop-types';
import { createStructuredSelector } from 'reselect';
import { Redirect } from 'react-router-dom';
import { getUser } from '../../selectors';
import authenticate from '../../actions/authAction';
import { scrollTop } from '../../utils/helpers';
import Profile from '../views/Profile';


/**
 * Container component for the Profile view
 * @export
 * @class ProfileContainer
 * @extends {Component}
 */
class ProfileContainer extends Component {
  handleEditProfile = async (event) => {
    event.preventDefault();
    const payload = formDataJSON(new FormData(event.target));
    payload.route = `users/${payload.userId}/profile`;
    payload.method = 'patch';
    const { editProfile: processEditProfile } = this.props;
    await processEditProfile(payload);
    scrollTop();
  }

  handleChangePassword = async (event) => {
    event.preventDefault();
    const payload = formDataJSON(new FormData(event.target));
    payload.route = 'users/password';
    payload.method = 'patch';
    scrollTop();
  }

  render() {
    const { user } = this.props;
    if (!user.isLoggedIn) return <Redirect to="/login" />;
    return (
      <Profile
      handleEditProfile={this.handleEditProfile}
      handleChangePassword={this.handleChangePassword}
      user={user}
      />
    );
  }
}
ProfileContainer.propTypes = {
  user: PropTypes.object.isRequired,
  editProfile: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector(
  {
    user: getUser
  }
);

const mapDispatchToProps = dispatch => ({
  editProfile: payload => dispatch(authenticate(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ProfileContainer);
