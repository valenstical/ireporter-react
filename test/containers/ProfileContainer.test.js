import React from 'react';
import setup from '../setup';
import ProfileContainer from '../../src/components/containers/ProfileContainer';


describe('<ProfileContainer>', () => {
  it('should render without crashing', () => {
    const wrapper = setup(<ProfileContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
