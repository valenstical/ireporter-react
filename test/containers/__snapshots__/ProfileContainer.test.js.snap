// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileContainer> should render without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "REPLACE",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/login",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(ProfileContainer)>
        <ProfileContainer
          editProfile={[Function]}
          user={
            Object {
              "isBusy": false,
              "message": Array [],
              "success": false,
            }
          }
        >
          <Redirect
            push={false}
            to="/login"
          />
        </ProfileContainer>
      </Connect(ProfileContainer)>
    </Router>
  </BrowserRouter>
</Provider>
`;
